CMakeLists.txt

```diff
@@ -37,7 +37,8 @@ set(${PROJECT_NAME}_SRCS
   src/meta_object.cpp
   src/multi_library_class_loader.cpp
 )
-add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS})
+file(GLOB ${PROJECT_NAME}_HDRS "include/${PROJECT_NAME}/*.h") # allow headers to be shown in CodeBlocks/QtCreator
```

Comment:
I'd advocate not to do it here given that no other package in the codebase aggregates headers globally for IDE purposes.